src/browser/reset.js

```diff
@@ -57,9 +60,9 @@ export async function reset(opts = {}) {
 		currentRtl = opts.rtl;
 	}
 
-	if (opts.lang !== currentLang) {
-		document.documentElement.setAttribute('lang', opts.lang);
-		currentLang = opts.lang;
+	opts.lang ??= '';
+	if (documentLocaleSettings.lamguage !== opts.lang) {
+		document.documentElement.lang = opts.lang;
```

Comment:
63. Avoids coercion of nullish values to strings
64. Compares against `documentLocaleSettings` for guaranteed latest change
65. Sets `document.documentElement.lang` to guarantee all values are in sync (`documentLocaleSettings` is observing this value)